File: solution/24039/Main.java
import java.io.*;

public class Main {
    static BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in), 1048576);
    static StringBuilder stdout = new StringBuilder();
    public static void main(String[] args) throws Exception {
        int N = Integer.parseInt(stdin.readLine());
        int prev = 2, cur = 3;
        for (int i = 4; prev * cur <= N; ++i) {
            boolean is_prime = true;
            for (int j = 2; j * j <= i; ++j) {
                if (i % j == 0) {
                    is_prime = false;
                    break;
                }
            }
            if (is_prime) {
                prev = cur;
                cur = i;
            }
        }
        stdout.append(prev * cur);
        System.out.println(stdout);
    }
}

File: solution/24039/main.py
def main():
    input = open(0, "rb").readline
    N = int(input())
    prev, cur, i = 2, 3, 4
    while prev * cur <= N:
        j = 2
        while j * j <= i:
            if i % j == 0:
                break
            j += 1
        else:
            prev = cur
            cur = i
        i += 1
    print(prev * cur)


if __name__ == "__main__":
    main()
